packages/core-data/src/resolvers.js

```diff
@@ -799,23 +835,35 @@ export const getDefaultTemplateId =
 		// Wait for the the entities config to be loaded, otherwise receiving
 		// the template as an entity will not work.
 		await resolveSelect.getEntitiesConfig( 'postType' );
+		const id = template?.wp_id || template?.id;
```

Comment:
I'm a huge fan of this PR, thank you!

I'm assuming `wp_id` property will longer be attached to (customized) template revisions since it's inheriting the post controller?

Can be another PR, but I think for the revisions API to work here the [default id should now be used as the key](https://github.com/WordPress/gutenberg/blob/42592862a355e19dcb1893cd4de2faf8de0df8b1/packages/core-data/src/entities.js#L357-L357). 

Currently, the template name must be used as the id e.g., `getRevisions( 'postType', 'wp_template', 'twentytwentyfive//home' )`

With this PR we can use the post id, which is great!

```diff
diff --git a/packages/core-data/src/entities.js b/packages/core-data/src/entities.js
index 3c10676750..6deb8267a3 100644
--- a/packages/core-data/src/entities.js
+++ b/packages/core-data/src/entities.js
@@ -354,7 +354,7 @@ async function loadPostTypeEntities() {
 				}/${ parentId }/revisions${
 					revisionId ? '/' + revisionId : ''
 				}`,
-			revisionKey: isTemplate ? 'wp_id' : DEFAULT_ENTITY_KEY,
+			revisionKey: DEFAULT_ENTITY_KEY,
 		};
 	} );
 }
```

To test that it works:

```
await wp.data.resolveSelect( 'core' ).getRevisions( 'postType', 'wp_template', templatePostId )

await wp.data.resolveSelect( 'core' ).getRevision( 'postType', 'wp_template', templatePostId, revisionId )
```

Not sure if there are implications for sites with existing template revisions.

Reply:
Thanks! Yes, there's probably a bunch of improvements that I overlooked. I'll address this on.